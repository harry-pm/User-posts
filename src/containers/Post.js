import React, { Component } from 'react'
import Axios from 'axios';
import Comment from '../components/Comment';

class Post extends Component {

  state = {
      comments: [],
      newComment: ''
  };

  getComments () {
      Axios.get(`https://jsonplaceholder.typicode.com/comments?postId=${this.props.post.id}`).then(res => {
          this.setState({comments: res.data});
      })
  }

  componentDidMount() {
      this.getComments();
  }

  // handleChange(e) {
  //   this.setState({newComment: e.target.value})
  //   console.log('e.target.value:', e.currentTarget.value)
  // }

  // handleSubmit(e) {
  //   e.preventDefault();
  //   const x = [...this.state.comments, this.state.newComment]
  //   this.setState({comments: x})
  // }

  render() {
    return (
      <>
        <h4>{this.props.post.title}</h4>
        <p>{this.props.post.body}</p>
        <h5>comments</h5>
        <h6>Add comment</h6>
        {/* <form action="" onSubmit={this.handleSubmit}>
            <input type="text" name="" id="" onChange={this.handleChange} />
            <input type="submit" value="Submit"/>
        </form> */}
        {this.state.comments.map(comment => (
            <Comment comment={comment} key={comment.id}/>
        ))}
      </>
    )
  }
}

export default Post