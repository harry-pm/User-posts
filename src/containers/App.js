import React from 'react';
import Post from './Post';
import axios from 'axios';
import About from '../components/About';
import Title from '../components/Title';
import {
    Route,
    Link,
    BrowserRouter as Router,
    Switch
} from 'react-router-dom';

class App extends React.Component {

    state = {
        posts: []
    };

    getPosts () {
        axios.get('https://jsonplaceholder.typicode.com/posts').then((res) => {
            this.setState({posts: res.data});
        })
    };

    componentDidMount() {
        this.getPosts()
    };


    render(){
        return (
            <div className='App'>
                <Router>
                    <Link exact='true' to="/"><strong>Home</strong></Link> |&nbsp;
                    <Link to="/about"><strong>About</strong></Link>
                    {this.state.posts.map(post => (
                        <React.Fragment key={post.id}>
                            <Switch>
                                <Route exact path="/">
                                    <Link to={`/${post.id}`}>
                                        <Title title={post.title} />
                                    </Link>
                                </Route>
                            </Switch>
                            <Route path={`/${post.id}`}>
                                <Post className='post' post={post}/>
                            </Route>
                        </React.Fragment>
                    ))}
                    <Route path={`/about`}>
                        <About />
                    </Route>
                </Router>
            </div>
        )
    }
}

export default App;
