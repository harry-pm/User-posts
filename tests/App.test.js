import React from 'react';
import App from '../src/containers/App';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import {expect as exp} from 'chai';
import Post from '../src/containers/Post';

describe('App', () => {
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer
        .create(<App />).toJSON()
        expect(tree).toMatchSnapshot();
    })

    it('renders Posts', () => {
        const posts = [{id: '2', title: 'hi', body: 'hello'}]
        const postComponent = shallow(<App post={post}/>)
        expect(postComponent).toMatchSnapshot();
    })
})

//shallow does a fake browser to compare with
//mount - where shallow only goes one level deep, mount will go recursivley and render all the children

