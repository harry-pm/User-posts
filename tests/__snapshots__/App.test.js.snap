// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <a
    exact="true"
    href="/"
    onClick={[Function]}
  >
    <strong>
      Home
    </strong>
  </a>
   |Â 
  <a
    href="/about"
    onClick={[Function]}
  >
    <strong>
      About
    </strong>
  </a>
</div>
`;
